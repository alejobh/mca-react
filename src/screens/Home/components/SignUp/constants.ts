/* eslint-disable @typescript-eslint/naming-convention */
import i18next from 'i18next';

export const VALIDATIONS = {
  EMAIL: {
    required: i18next.t('SignUp:required') as string,
    pattern: {
      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
      message: i18next.t('SignUp:invalidFormat') as string
    }
  },
  PASSWORD: {
    required: i18next.t('signUp:required') as string,
    pattern: {
      value: /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[^\w\s\d])?.+$/,
      message: i18next.t('SignUp:validFormatPassword') as string
    },
    minLength: {
      value: 8,
      message: i18next.t('SignUp:minLenghtPassword') as string
    }
  }
};
