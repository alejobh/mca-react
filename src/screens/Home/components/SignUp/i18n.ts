import i18next from 'i18next';

i18next.addResources('es', 'SignUp', {
  firstName: 'Nombre',
  lastName: 'Apellido',
  email: 'Email',
  passwordConfirmation: 'Confirmaci√≥n de Password'
});
