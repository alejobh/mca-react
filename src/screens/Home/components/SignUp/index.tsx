/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { useForm } from 'react-hook-form';
import i18next from 'i18next';

import WoloxLogo from '../../assets/LogoWolox-Original.png';

import styles from './styles.module.scss';

type InputsValues = {
  email: string;
  password: string;
  password_confirmation: string;
  first_name: string;
  last_name: string;
  locale: string;
};

enum UserFieldIds {
  FIRST_NAME = 'first_name',
  LAST_NAME = 'last_name',
  EMAIL = 'email',
  PASSWORD = 'password',
  PASSWORD_CONFIRMATION = 'password_confirmation'
}

function SignUp() {
  const { register, handleSubmit } = useForm<InputsValues>();
  const onSubmit = (data: InputsValues) => console.log(data);

  return (
    <form className={styles.signUpForm} onSubmit={handleSubmit(onSubmit)}>
      <img src={WoloxLogo} className={`m-bottom-3 ${styles.logo}`} />
      <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
        <label htmlFor={UserFieldIds.FIRST_NAME} className={`m-bottom-2 ${styles.label}`}>
          {i18next.t('SignUp:firstName')}
        </label>
        <input {...register(UserFieldIds.FIRST_NAME)} className={styles.input} />
      </div>
      <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
        <label htmlFor={UserFieldIds.LAST_NAME} className={`m-bottom-2 ${styles.label}`}>
          {i18next.t('SignUp:lastName')}
        </label>
        <input {...register(UserFieldIds.LAST_NAME)} className={styles.input} />
      </div>
      <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
        <label htmlFor={UserFieldIds.EMAIL} className={`m-bottom-2 ${styles.label}`}>
          {i18next.t('SignUp:email')}
        </label>
        <input type="email" {...register(UserFieldIds.EMAIL)} className={styles.input} autoComplete="off" />
      </div>
      <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
        <label htmlFor={UserFieldIds.PASSWORD} className={`m-bottom-2 ${styles.label}`}>
          Password
        </label>
        <input type="password" {...register(UserFieldIds.PASSWORD)} className={styles.input} />
      </div>
      <div className={`column start m-bottom-5 ${styles.inputContainer}`}>
        <label htmlFor={UserFieldIds.PASSWORD_CONFIRMATION} className={`m-bottom-2 ${styles.label}`}>
          {i18next.t('SignUp:passwordConfirmation')}
        </label>
        <input type="password" {...register(UserFieldIds.PASSWORD_CONFIRMATION)} className={styles.input} />
      </div>
      <button type="submit" className={`m-bottom-3 ${styles.signUpButton}`} />
      <div className={`m-bottom-3 ${styles.line}`} />
      <button type="button" className={styles.signUpButtonSecondary} />
    </form>
  );
}

export default SignUp;
