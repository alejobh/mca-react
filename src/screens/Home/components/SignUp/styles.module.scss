@import '../../../../scss/variables/colors';

.sign-up-form {
  align-items: center;
  background-color: $white-smoke;
  border-top: 8px solid $cian;
  display: flex;
  flex-direction: column;
  max-width: 300px;
  padding: 28px 25px 24px;
  width: 100%;
}

.logo {
  height: 70px;
  width: 212px;
  object-fit: contain;
}

.input-container {
  width: 100%;
}

.input {
  height: 32px;
  width: 100%;
}

.label {
  color: $black;
  font-size: 16px;
  line-height: 20px;
}

.sign-up-button {
  background-image: url('../../assets/sign-up-button.png');
  height: 32px;
  width: 100%;

  &-secondary {
    background-image: url('../../assets/login-button.png');
    height: 32px;
    width: 100%;
  }
}

.line {
  background-color: $light-grey;
  height: 2px;
  width: 100%;
}
