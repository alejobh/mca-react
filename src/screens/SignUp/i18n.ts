import i18next from 'i18next';

i18next.addResources('es', 'SignUp', {
  signUpTitle: 'Sign Up',
  firstName: 'Nombre',
  lastName: 'Apellido',
  required: 'Este campo es requerido',
  email: 'Email',
  invalidFormat: 'Formato inválido',
  password: 'Contraseña',
  passwordConfirmation: 'Confirmación de Contraseña',
  validFormatPassword: 'Debe contener 1 mayúscula, 1 minúscula y 1 número',
  minLenghtPassword: 'Mínimo 8 caracteres',
  noMatchPassword: 'Las contraseñas no coinciden',
  messageError: 'Error al crear usuario'
});
