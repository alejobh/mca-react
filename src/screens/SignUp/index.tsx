/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable @typescript-eslint/no-unused-vars */
import { useForm } from 'react-hook-form';
import { useMutation } from 'react-query';
import i18next from 'i18next';
import { useState } from 'react';

import WoloxLogo from 'assets/LogoWolox-Original.png';
import { signUp } from 'services/UserService';
import Loading from 'components/Spinner/components/loading';

import { UserForm } from '../../interfaces/interfaces';

import styles from './styles.module.scss';

enum UserFieldIds {
  FIRST_NAME = 'first_name',
  LAST_NAME = 'last_name',
  EMAIL = 'email',
  PASSWORD = 'password',
  PASSWORD_CONFIRMATION = 'password_confirmation'
}

function SignUp() {
  const {
    register,
    handleSubmit,
    formState: { errors },
    watch
  } = useForm<UserForm>();
  const [notCredentials, setNotCredentials] = useState(false);

  const mutation = useMutation(signUp, {
    onError: () => {
      setNotCredentials(true);
    },
    onSuccess: data => {
      console.log(data);
    }
  });

  const onSubmit = (user: UserForm) => mutation.mutate(user);

  return (
    <div className="column center middle full-width full-height">
      <form className={`m-bottom-2 ${styles.signUpForm}`} onSubmit={handleSubmit(onSubmit)}>
        <img
          src={WoloxLogo}
          alt={`${i18next.t('Global:logoWolox')}`}
          className={`m-bottom-3 ${styles.logo}`}
        />
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label
            htmlFor={UserFieldIds.FIRST_NAME}
            className={`m-bottom-2 ${styles.label}`}
            id="firstname-label"
          >
            {i18next.t('SignUp:firstName')}
          </label>
          <input
            aria-labelledby="firstname-label"
            {...register(UserFieldIds.FIRST_NAME, {
              required: i18next.t('SignUp:required') as string
            })}
            className={styles.input}
          />
          {errors.first_name && <span className={styles.inputError}>{errors.first_name.message}</span>}
        </div>
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label
            htmlFor={UserFieldIds.LAST_NAME}
            className={`m-bottom-2 ${styles.label}`}
            id="lastname-label"
          >
            {i18next.t('SignUp:lastName')}
          </label>
          <input
            aria-labelledby="lastname-label"
            {...register(UserFieldIds.LAST_NAME, {
              required: i18next.t('SignUp:required') as string
            })}
            className={styles.input}
          />
          {errors.last_name && <span className={styles.inputError}>{errors.last_name.message}</span>}
        </div>
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label htmlFor={UserFieldIds.EMAIL} className={`m-bottom-2 ${styles.label}`} id="email-label">
            {i18next.t('SignUp:email')}
          </label>
          <input
            aria-labelledby="email-label"
            type="email"
            {...register(UserFieldIds.EMAIL, {
              required: i18next.t('SignUp:required') as string,
              pattern: {
                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                message: i18next.t('SignUp:invalidFormat') as string
              }
            })}
            className={styles.input}
          />
          {errors.email && <span className={styles.inputError}>{errors.email.message}</span>}
        </div>
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label htmlFor={UserFieldIds.PASSWORD} className={`m-bottom-2 ${styles.label}`} id="password-label">
            {i18next.t('SignUp:password') as string}
          </label>
          <input
            aria-labelledby="password-label"
            type="password"
            {...register(UserFieldIds.PASSWORD, {
              required: i18next.t('SignUp:required') as string,
              pattern: {
                value: /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[^\w\s\d])?.+$/,
                message: i18next.t('SignUp:validFormatPassword') as string
              },
              minLength: {
                value: 8,
                message: i18next.t('SignUp:minLenghtPassword') as string
              }
            })}
            className={styles.input}
          />
          {errors.password && <span className={styles.inputError}>{errors.password.message}</span>}
        </div>
        <div className={`column start m-bottom-5 ${styles.inputContainer}`}>
          <label
            htmlFor={UserFieldIds.PASSWORD_CONFIRMATION}
            className={`m-bottom-2 ${styles.label}`}
            id="password-confirmation-label"
          >
            {i18next.t('SignUp:passwordConfirmation')}
          </label>
          <input
            aria-labelledby="password-confirmation-label"
            type="password"
            {...register(UserFieldIds.PASSWORD_CONFIRMATION, {
              required: true,
              validate: value =>
                value === watch(UserFieldIds.PASSWORD) || (i18next.t('SignUp:noMatchPassword') as string)
            })}
            className={styles.input}
          />
          {errors.password_confirmation && (
            <span className={styles.inputError}>{errors.password_confirmation.message}</span>
          )}
        </div>
        <button type="submit" className={`m-bottom-3 ${styles.signUpButton}`}>
          {mutation.isLoading ? <Loading /> : i18next.t('SignUp:signUpTitle')}
        </button>
        <div className={`m-bottom-3 ${styles.line}`} />
        <button type="button" className={styles.signUpButtonSecondary} />
      </form>
      {notCredentials && <span className={styles.error}>{i18next.t('SignUp:messageError')}</span>}
    </div>
  );
}

export default SignUp;
