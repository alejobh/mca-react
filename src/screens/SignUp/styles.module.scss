@import '../../scss/variables/colors';

@import '../../scss/variables/texts';

.sign-up-form {
  align-items: center;
  background-color: $white-smoke;
  border-top: 8px solid $cian;
  display: flex;
  flex-direction: column;
  max-width: 300px;
  padding: 28px 25px 24px;
  width: 100%;
}

.logo {
  height: 70px;
  width: 212px;
  object-fit: contain;
}

.input-container {
  width: 100%;
}

.input {
  height: 32px;
  width: 100%;
}

.input-error {
  color: $red;
  font-size: $font-size-small;
}

.label {
  color: $black;
  font-size: $font-size-medium;
  line-height: 20px;
}

.sign-up-button {
  background-color: $yellow-green;
  border-radius: 5px;
  color: $white;
  font-size: $font-size-medium;
  height: 32px;
  width: 100%;

  &-secondary {
    background-image: url('../../assets/login-button.png');
    height: 32px;
    width: 100%;
  }
}

.line {
  background-color: $light-grey;
  height: 2px;
  width: 100%;
}

.no-error {
  display: none;
}

.error {
  color: $red;
  font-size: $font-size-large;
}
