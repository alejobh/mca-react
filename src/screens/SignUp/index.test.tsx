import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from 'react-query';

import SignUp from './index';

const queryClient = new QueryClient();

describe('SignUp screen', () => {
  test('matches last snapshot', () => {
    const component = render(
      <QueryClientProvider client={queryClient}>
        <SignUp />
      </QueryClientProvider>
    );

    expect(component).toMatchSnapshot();
    userEvent.type(screen.getByLabelText('SignUp:firstName'), 'Test Name');
    userEvent.type(screen.getByLabelText('SignUp:lastName'), 'Test LastName');
    userEvent.type(screen.getByLabelText('SignUp:email'), 'test@gmail.com');
    userEvent.type(screen.getByLabelText('SignUp:password'), 'wolox1189');
    userEvent.type(screen.getByLabelText('SignUp:passwordConfirmation'), 'wolox1189');
  });
});
