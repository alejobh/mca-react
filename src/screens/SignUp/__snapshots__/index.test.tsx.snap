// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp screen matches last snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="column center middle full-width full-height"
      >
        <form
          class="m-bottom-2 signUpForm"
        >
          <img
            alt="Global:logoWolox"
            class="m-bottom-3 logo"
            src="LogoWolox-Original.png"
          />
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="first_name"
              id="firstname-label"
            >
              SignUp:firstName
            </label>
            <input
              aria-labelledby="firstname-label"
              class="input"
              name="first_name"
            />
          </div>
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="last_name"
              id="lastname-label"
            >
              SignUp:lastName
            </label>
            <input
              aria-labelledby="lastname-label"
              class="input"
              name="last_name"
            />
          </div>
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="email"
              id="email-label"
            >
              SignUp:email
            </label>
            <input
              aria-labelledby="email-label"
              class="input"
              name="email"
              type="email"
            />
          </div>
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="password"
              id="password-label"
            >
              SignUp:password
            </label>
            <input
              aria-labelledby="password-label"
              class="input"
              name="password"
              type="password"
            />
          </div>
          <div
            class="column start m-bottom-5 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="password_confirmation"
              id="password-confirmation-label"
            >
              SignUp:passwordConfirmation
            </label>
            <input
              aria-labelledby="password-confirmation-label"
              class="input"
              name="password_confirmation"
              type="password"
            />
          </div>
          <button
            class="m-bottom-3 signUpButton"
            type="submit"
          >
            SignUp:signUpTitle
          </button>
          <div
            class="m-bottom-3 line"
          />
          <button
            class="signUpButtonSecondary"
            type="button"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="column center middle full-width full-height"
    >
      <form
        class="m-bottom-2 signUpForm"
      >
        <img
          alt="Global:logoWolox"
          class="m-bottom-3 logo"
          src="LogoWolox-Original.png"
        />
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="first_name"
            id="firstname-label"
          >
            SignUp:firstName
          </label>
          <input
            aria-labelledby="firstname-label"
            class="input"
            name="first_name"
          />
        </div>
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="last_name"
            id="lastname-label"
          >
            SignUp:lastName
          </label>
          <input
            aria-labelledby="lastname-label"
            class="input"
            name="last_name"
          />
        </div>
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="email"
            id="email-label"
          >
            SignUp:email
          </label>
          <input
            aria-labelledby="email-label"
            class="input"
            name="email"
            type="email"
          />
        </div>
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="password"
            id="password-label"
          >
            SignUp:password
          </label>
          <input
            aria-labelledby="password-label"
            class="input"
            name="password"
            type="password"
          />
        </div>
        <div
          class="column start m-bottom-5 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="password_confirmation"
            id="password-confirmation-label"
          >
            SignUp:passwordConfirmation
          </label>
          <input
            aria-labelledby="password-confirmation-label"
            class="input"
            name="password_confirmation"
            type="password"
          />
        </div>
        <button
          class="m-bottom-3 signUpButton"
          type="submit"
        >
          SignUp:signUpTitle
        </button>
        <div
          class="m-bottom-3 line"
        />
        <button
          class="signUpButtonSecondary"
          type="button"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
